### PR TITLE
Upgrade scalafmt but keep defaults from 2.1.0

### PR DESCRIPTION
**Pull request by chbatey**
The `edition`, to preserve existing behavior while upgrading to latest for perf upgrades, mainly works but some bugs have been fixed e.g.
https://github.com/scalameta/scalafmt/issues/1509 that cause 
formatting changes

For me this makes clean;scalafmtAll 3x faster

### CODE REVIEW

**Review by raboof — APPROVED** [review]
Seems worth it for the speed increase

File: akka-actor-tests/src/test/scala/akka/routing/MetricsBasedResizerSpec.scala [hunk 1]
```diff
@@ -134,9 +134,8 @@ class MetricsBasedResizerSpec extends AkkaSpec(ResizerSpec.config) with DefaultT
 
     "stop an underutilizationStreak when fully utilized" in {
       val resizer = DefaultOptimalSizeExploringResizer()
-      resizer.record = ResizeRecord(
-        underutilizationStreak =
-          Some(UnderUtilizationStreak(start = LocalDateTime.now.minusHours(1), highestUtilization = 1)))
+      resizer.record = ResizeRecord(underutilizationStreak =
```
Comment (jrudolph) [inline akka-actor-tests/src/test/scala/akka/routing/MetricsBasedResizerSpec.scala]: Can we undo this change with a setting? Doesn't seem better now.

**Review by jrudolph — COMMENTED** [review]
The `else` change seems good but the named parameter thing is weird.

**Comment by chbatey** [comment]
I agree, will look into that

**Comment by ennru** [comment]
Worth to warm up, or just redo with the current versions?

**Comment by patriknw** [comment]
I guess this is stale
